docs: updates to overview & ReadMe pages (#232)
intel-retail/automated-self-checkout: docs_src/index.md

## Overview

> As Computer Vision is becoming more and more popular especially, for industrial & retail use cases, development and deployment of these solutions are becoming more challenging. Vision workloads are large and complex and need to go through many stages as shown in the figure below -

Reviewer: ```suggestion
As Computer Vision is becoming more and more popular, especially for industrial & retail use cases, development and deployment of these solutions are becoming more challenging. Vision workloads are large and complex and need to go through many stages as shown in the figure below -
```

> As Computer Vision is becoming more and more popular, especially for industrial & retail use cases, development and deployment of these solutions are becoming more challenging. Vision workloads are large and complex and need to go through many stages as shown in the figure below -

Reviewer 2: ```suggestion
As Computer Vision becomes more and more mainstream, especially for industrial & retail use cases, development and deployment of these solutions becomes more challenging. Vision workloads are large and complex and need to go through many stages as shown in the figure below.
```

[![Vision Data Flow](./images/vision-data-flow.jpg)](./images/vision-data-flow.jpg)

> Automated self-checkout solutions are complex, and retailers, independent software vendors (ISVs), and system integrators (SIs) require a good understanding of hardware and software, the costs involved in setting up and scaling the system, and the configuration that best suits their needs. Vision workloads are significantly larger and require systems to be architected, built, and deployed with several considerations. Hence, a set of ingredients needed to create an automated self-checkout solution is necessary. More details on this use case are available on the [Intel Developer Focused Webpage](https://www.intel.com/content/www/us/en/developer/articles/reference-implementation/automated-self-checkout.html) and on this [LinkedIn Blog](https://www.linkedin.com/pulse/retail-innovation-unlocked-open-source-vision-enabled-mohideen/)

Reviewer 2: ```suggestion
Automated self-checkout solutions are complex; retailers, independent software vendors (ISVs), and system integrators (SIs) require a good understanding of hardware and software, the costs involved in setting up and scaling the system, and the configuration that best suits their needs. Vision workloads are significantly larger and require systems to be architected, built, and deployed with several considerations. Hence, automated self-checkout solutions require customizability to fit the use case. More details are available on the [Intel Developer Focused Webpage](https://www.intel.com/content/www/us/en/developer/articles/reference-implementation/automated-self-checkout.html) and on this [LinkedIn Blog](https://www.linkedin.com/pulse/retail-innovation-unlocked-open-source-vision-enabled-mohideen/)
```

Author: these are mostly from the blog/developer page, won't make much changes.
